tools/jsonstats.cpp

```diff
@@ -133,12 +183,32 @@ int main(int argc, char *argv[]) {
     return EXIT_FAILURE;
   }
 
-  printf("# integer_count float_count string_count backslash_count "
-         "non_ascii_byte_count object_count array_count null_count true_count "
-         "false_count byte_count structural_indexes_count\n");
-  printf("%zu %zu %zu %zu %zu %zu %zu %zu %zu %zu %zu %zu\n", s.integer_count,
-         s.float_count, s.string_count, s.backslash_count,
-         s.non_ascii_byte_count, s.object_count, s.array_count, s.null_count,
-         s.true_count, s.false_count, s.byte_count, s.structural_indexes_count);
-  return EXIT_SUCCESS;
+  printf(R"({
+      "integer_count"            = %10zu,
+      "float_count"              = %10zu,
+      "string_count"             = %10zu,
+      "ascii_string_count"       = %10zu,
+      "backslash_count"          = %10zu,
+      "non_ascii_byte_count"     = %10zu,
+      "object_count"             = %10zu,
+      "maximum_object_size"      = %10zu,
+      "array_count"              = %10zu,
+      "maximum_array_size"       = %10zu,
+      "null_count"               = %10zu,
+      "true_count"               = %10zu,
+      "false_count"              = %10zu,
+      "byte_count"               = %10zu,
+      "structural_indexes_count" = %10zu,
+      "key_count"                = %10zu,
+      "ascii_key_count"          = %10zu,
+      "key_maximum_length"       = %10zu,
+      "maximum_depth"            = %10zu
 }
+)",
+         s.integer_count, s.float_count, s.string_count, s.ascii_string_count,
```

Comment:
It's a pain in terms of line count, but this will be easier to review and read if each of the actual variables are on their own line--that way you can scan both of them for order differences a little easier. The order appears right, though!

Reply:
I don't think that either C++ or C offer a good solution to this problem. In Swift, this would not be an issue at all.